### PR TITLE
Enable @vectorize for target dppy

### PR DESCRIPTION
**Pull request by reazulhoque**
This PR will allow users to decorate functions with @vectorize which can be directly offloaded to target dppy.

This PR is a part of #277 and supercedes https://github.com/IntelPython/numba-dppy/pull/289.

### CODE REVIEW

File: numba_dppy/examples/vectorize.py [hunk 1]
```diff
@@ -55,5 +55,21 @@ def main():
...
+@vectorize([float64(float64, float64)], target="dppy")
+def vector_add(a, b):
+    return a + b
+
+
+def test_vectorize():
+    A = np.arange(10, dtype=np.float64).reshape((5, 2))
+    B = np.arange(10, dtype=np.float64).reshape((5, 2))
+
+    with dpctl.device_context("opencl:gpu:0"):
```
Comment (diptorupd) [inline numba_dppy/examples/vectorize.py:67]: Can we make the test parameterized as usual? Or maybe just `default_context`?

File: numba_dppy/dpnp_glue/dpnp_array_creations_impl.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def dpnp_ones_like_impl(a, dtype=None):
 
     """
     res_dtype = dtype
-    if dtype == types.none:
+    if dtype == types.none or dtype is None:
```
Comment (diptorupd) [inline numba_dppy/dpnp_glue/dpnp_array_creations_impl.py:130]: Why are these changes in this PR? Are the related to the @vectorize work?

File: numba_dppy/examples/vectorize.py [hunk 3]
```diff
@@ -55,5 +55,21 @@ def main():
     print("Done...")
 
 
+@vectorize([float64(float64, float64)], target="dppy")
```
Comment (diptorupd) [inline numba_dppy/examples/vectorize.py:58]: Is not the `target` keyword deprecated in Numba?

File: numba_dppy/vectorizers.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
+# Copyright 2021 Intel Corporation
```
Comment (diptorupd) [inline numba_dppy/vectorizers.py:1]: Let us follow the same naming convention as Numba, *i.e*, `numba_dppy.np.ufunc.decorators`.  Also, few places have missing docstrings including the module level docstring.

File: numba_dppy/dpnp_glue/dpnp_array_creations_impl.py (hunk 2 above)
Comment (reazulhoque) [reply numba_dppy/dpnp_glue/dpnp_array_creations_impl.py:130]: They are not, but they were discovered during the work somehow.

File: numba_dppy/examples/vectorize.py (hunk 3 above)
Comment (reazulhoque) [reply numba_dppy/examples/vectorize.py:58]: I do not think it is deprecated for `@vectorize`. Numba master still has this test for Cuda: https://github.com/numba/numba/blob/master/numba/cuda/tests/cudapy/test_vectorize_decor.py#L27

File: numba_dppy/vectorizers.py (hunk 4 above)
Comment (reazulhoque) [reply numba_dppy/vectorizers.py:1]: It was followed using, cuda in Numba: https://github.com/numba/numba/blob/master/numba/cuda/vectorizers.py

**Comment by PokhodenkoSA** [comment]
@reazulhoque I did not finish yet. Please, skip tests if GPU devices are not present. See this log: https://github.com/reazulhoque/numba-dppy/runs/3164195876#step:10:2206

File: numba_dppy/vectorizers.py (hunk 4 above)
Comment (PokhodenkoSA) [reply numba_dppy/vectorizers.py:1]: Lets keep it simple as is.

**Review by PokhodenkoSA — APPROVED** [review]
LGTM.

**Pull request merged**